File: src/main/java/org/pl/lexer/tokenizer/StatementTokenizer.java
package org.pl.lexer.tokenizer;

import org.pl.lexer.ILexer;
import org.pl.lexer.token.StatementSeparatorToken;

public class StatementTokenizer implements ITokenizer {
    @Override
    public boolean matches(Character c) {
        return c == ';';
    }

    @Override
    public TokenizerResult tokenize(ILexer lexer) {
        return new TokenizerResult(new StatementSeparatorToken());
    }
}

File: src/main/java/org/pl/lexer/tokenizer/StatementTokenizer.kt
package org.pl.lexer.tokenizer

import org.pl.lexer.ILexer
import org.pl.lexer.token.StatementSeparatorToken

/**
 * Tokenizes the statement separation symbol ';' and '\n'.
 */
class StatementTokenizer : ITokenizer {
    override fun matches(c: Char) = c == ';' || c == '\n'
    override fun tokenize(lexer: ILexer) = StatementSeparatorToken()
}